File: app/src/main/java/mobiledev/erickgomes/picpayapp/database/CreditCardDao.java
package mobiledev.erickgomes.picpayapp.database;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import mobiledev.erickgomes.picpayapp.models.CreditCard;

/**
 * Created by erickgomes on 24/03/2018.
 */

@Dao
public interface CreditCardDao {

    @Query("SELECT * FROM creditcards")
    List<CreditCard> getAllCreditCards();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public void insertCreditCard(CreditCard creditCard);
}

File: picpay/app/src/main/java/br/com/picpay/picpay/db/CreditCardDao.kt
package br.com.picpay.picpay.db

import android.arch.persistence.room.*
import io.reactivex.Single

@Dao
interface CreditCardDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun save (creditCard: CreditCard)

    @Query("DELETE FROM credit_card")
    fun remove ()

    @Query("SELECT * FROM credit_card")
    fun getCreditCard(): Single<List<CreditCard>>
}